source/fundamentals/enterprise-auth.txt

```diff
@@ -1,27 +1,192 @@
+   }
+
+   uri := "mongodb://<hostname>:<port>"
+   clientOpts := options.Client().ApplyURI(uri).SetAuth(credential)
+   
+   client, err := mongo.Connect(context.TODO(), clientOpts)
+
+Configuration Options
+~~~~~~~~~~~~~~~~~~~~~
+
+You can specify additional properties with your authentication
+mechanism using the ``AuthMechanismProperties`` field in the
+``Credential`` struct. The following code shows how you can set values
+for the ``SERVICE_NAME`` and ``SERVICE_REALM`` fields when defining a
+``Credential`` struct:
+
+.. code-block:: go
+
+   credential := options.Credential{
+           AuthMechanism: "GSSAPI",
+           Username:      "<username>",
+           Password:      "<password>",
+           AuthMechanismProperties: map[string]string{
+                   "SERVICE_REALM": "<Kerberos service realm>",
+                   "SERVICE_NAME":  "<service name>",
+           },
+   }
+
+For a full list of properties, see the
+:manual:`Server manual entry on authentication properties </reference/connection-string/#mongodb-urioption-urioption.authMechanismProperties>`.
+
+.. _golang-LDAP:
+
+Authenticate to LDAP (PLAIN)
+----------------------------
+
+You can authenticate to a Lightweight Directory Access Protocol (LDAP) server 
+using your directory server username and password.
+
+.. warning::
+
+   This authentication mechanism sends the password to the server in
+   plaintext, so you should use this mechanism only with TLS connections.
+
+The :ref:`example <golang-LDAP-example>` specifies the authentication mechanism using the following
+placeholders:
+
+- ``username``: Your LDAP username
+- ``password``: Your LDAP password
+- ``hostname``: The network address of your MongoDB server, accessible
+  by your client
+- ``port``: Your MongoDB server's port number
+
+.. _golang-LDAP-example:
+
+Example
+~~~~~~~
+
+The following code shows how you can define a ``Credential`` struct to
+authenticate to LDAP and create a client with your authentication
+preferences:
+
+.. code-block:: go
+
+   credential := options.Credential{
+           AuthMechanism: "PLAIN",
+           Username:      "<username>",
+           Password:      "<password>",
+	}
+
+   uri := "mongodb://<hostname>:<port>"
+   clientOpts := options.Client().ApplyURI(uri).SetAuth(credential)
+   
+   client, err := mongo.Connect(context.TODO(), clientOpts)
+
+.. tip::
+
+   The method refers to PLAIN instead of LDAP since it
+   authenticates using the PLAIN Simple Authentication and Security Layer
+   (SASL) defined in `RFC-4616 <https://tools.ietf.org/html/rfc4616>`__.
+
+Additional Information
+----------------------
+
+To learn more about the concepts in this guide, see the following documentation:
+
+- :manual:`Kerberos Authentication </core/kerberos/>`
+- :manual:`LDAP Proxy Authentication </core/security-ldap/>`
+- :ref:`<golang-connection-guide>`
+- :ref:`<golang-authentication-mechanisms>`
+- :manual:`Connection String URI Format </reference/connection-string/>`
```

Comment:
Suggestion:

I think the Go driver docs is more relevant (unless you've linked to it already):
https://www.mongodb.com/docs/drivers/go/current/fundamentals/connection/

Reply:
Looks like you already linked to the connection guide. To make it what I think would be easier and more approachable would be to follow the style guide's recommendations on describing what they contain and to what site they're going.